### PR TITLE
 Use actual file name casing

### LINKED ISSUE #4664: Output is not deterministic when link and the link article has different casing
Found during azure-docs-pr integration testing

```diff

--- "a/D:/Docs.Build.TestData\\output\\azure-docs-pr/.dependencymap.json"
+++ "b/D:/Docs.Build.TestData\\.temp\\azure-docs-pr/.dependencymap.json"
@@ -205444,7 +205444,7 @@
         "type": "bookmark"
       },
       {
-        "source": "articles/cognitive-services/LUIS/luis-how-to-azure-subscription.md",
+        "source": "articles/cognitive-services/luis/luis-how-to-azure-subscription.md",
         "type": "link"
       },
       {
@@ -215772,7 +215772,7 @@
         "type": "link"
       },
       {
-        "source": "articles/cognitive-services/Speech-Service/speech-devices-sdk.md",
+        "source": "articles/cognitive-services/speech-service/speech-devices-sdk.md",
         "type": "link"
       },
       {
diff --git "a/D:/Docs.Build.TestData\\output\\azure-docs-pr/.errors.log" "b/D:/Docs.Build.TestData\\.temp\\azure-docs-pr/.errors.log"
index f55ec0eb..0b05f14f 100644
--- "a/D:/Docs.Build.TestData\\output\\azure-docs-pr/.errors.log"
+++ "b/D:/Docs.Build.TestData\\.temp\\azure-docs-pr/.errors.log"
@@ -1889,6 +1889,13 @@ To fix this error either change the JSON to a JSON primitive value (e.g. string,
   "file": "articles/active-directory/saas-apps/korn-ferry-alp-tutorial.md",
   "line": 150
 }
+{
+  "message_severity": "info",
+  "code": "at-xref-not-found",
+  "message": "Cross reference not found: '@contoso.com'",
+  "file": "articles/active-directory/saas-apps/kronos-tutorial.md",
+  "line": 157
+}
 {
   "message_severity": "info",
   "code": "at-xref-not-found",
@@ -2351,13 +2358,6 @@ To fix this error either change the JSON to a JSON primitive value (e.g. string,
   "file": "articles/active-directory/saas-apps/phraseanet-tutorial.md",
   "line": 147
 }
-{
-  "message_severity": "info",
-  "code": "at-xref-not-found",
-  "message": "Cross reference not found: '@contoso.com'",
-  "file": "articles/active-directory/saas-apps/pingboard-tutorial.md",
-  "line": 176
-}
 {
   "message_severity": "info",
   "code": "at-xref-not-found",
@@ -2593,22 +2593,22 @@ To fix this error either change the JSON to a JSON primitive value (e.g. string,
   "message_severity": "info",
   "code": "at-xref-not-found",
   "message": "Cross reference not found: '@contoso.com'",
-  "file": "articles/active-directory/saas-apps/sansan-tutorial.md",
-  "line": 210
+  "file": "articles/active-directory/saas-apps/sap-fiori-tutorial.md",
+  "line": 316
 }
```

### PR DESCRIPTION
**Pull request by yufeih**
Fixes #4664 , BuildScope related change in #4682

### CODE REVIEW

File: docs/specs/casing.yml [hunk 1]
```diff
@@ -85,3 +85,37 @@ outputs:
   docs/A.json: |
     {"conceptual":"<p><a href=\"B\"></a></p>"}
   docs/B.json:
+---
+# Use real file casing in manifest
+inputs:
+  docfx.yml: |
+    files: '*'
+  a.md: '[](media/ImAGe.PNG) [](MEdia/ImAGe.PNG)'
```
Comment (OsmondJiang) [inline docs/specs/casing.yml]: if we already have the casing file name, why we need lower case the resolved urls?

Comment (yufeih) [reply docs/specs/casing.yml]: For SEO purpose, output URLs are canonicalized into lower case because Google considers URLs to be case-sensitive

Comment (OsmondJiang) [reply docs/specs/casing.yml]: that's a different behavior with v2 right? so we need first figure out what's the SEO purpose for v2 behavior.  Before this PR, I though we lower case all resolve urls is a compromise way

Comment (yufeih) [reply docs/specs/casing.yml]: v2 also lower case urls, search `ARIA-and-UI-Automation` in [source](https://github.com/MicrosoftDocs/edge-developer/edit/master/microsoft-edge/accessibility/build.md) and [result](https://review.docs.microsoft.com/en-us/microsoft-edge/accessibility/build)

**Pull request merged**